src/uct/ib/dc/dc_mlx5_ep.h

```diff
@@ -536,7 +540,9 @@ static inline struct mlx5_grh_av *uct_dc_mlx5_ep_get_grh(uct_dc_mlx5_ep_t *ep)
             if (ucs_unlikely(status != UCS_OK)) { \
                 if (((_ep)->dci != UCT_DC_MLX5_EP_NO_DCI) && \
                     !uct_dc_mlx5_iface_is_dci_rand(_iface)) { \
-                    ucs_assertv_always(uct_dc_mlx5_iface_dci_has_outstanding(_iface, (_ep)->dci), \
+                    ucs_assertv_always(uct_dc_mlx5_iface_dci_has_outstanding(_iface, \
+                                                                             (_ep)->dci) || \
+                                       !uct_dc_mlx5_iface_has_tx_resources(_iface) , \
```

Comment:
why it is needed ? maybe epplain in the comment in line 531-534?

Reply:
fixed this problem by another way